src/main/kotlin/systems/ajax/englishstudytelegrambot/service/WordService.kt

```diff
@@ -1,35 +1,70 @@
 ) : WordService {
 
-    override suspend fun save(wordSpelling: String, wordTranslate: String): Word {
-        val word =
-            Word(wordSpelling, wordTranslate, additionalInfoAboutWordService.findAdditionInfoAboutWord(wordSpelling))
-        log.info("full saving word {}", word)
-        return wordRepository.save(word)
+    override suspend fun saveNewWord(libraryName: String, telegramUserId: String, wordDto: WordDto): Word {
+        val libraryId = libraryRepository.getLibraryIdByPairLibraryNameAndTelegramUserId(libraryName, telegramUserId)
+        libraryId.takeIf {
+            !wordRepository.isWordBelongsToLibraryByWordSpelling(wordDto.spelling, libraryId)
+        }?.let {
+            return wordRepository.saveNewWord(
+                Word(
+                    spelling = wordDto.spelling,
+                    translate = wordDto.translate,
+                    libraryId = libraryId,
+                    additionalInfoAboutWord = additionalInfoAboutWordService.findAdditionInfoAboutWord(wordDto.spelling)
+                )
+            )
+        } ?: throw WordAlreadyPresentInLibraryException()
     }
 
-    override fun getWordByItsSpelling(wordSpelling: String): Word =
-        wordRepository.findById(wordSpelling).orElseThrow { throw WordNotFoundBySpendingExceptions() }
+    override fun updateWordTranslate(libraryName: String, telegramUserId: String, wordDto: WordDto): Word {
+        val libraryId = libraryRepository.getLibraryIdByPairLibraryNameAndTelegramUserId(libraryName, telegramUserId)
+        val wordId = wordRepository.getWordIdBySpellingAndLibraryId(wordDto.spelling, libraryId)
+        wordId.takeIf {
+            wordRepository.isWordBelongsToLibraryByWordId(wordId, libraryId)
+        }?.let {
+            return wordRepository.updateWordTranslating(wordId, wordDto.translate)
+        } ?: throw WordNotFoundBySpendingException()
+    }
+
+    override fun deleteWord(libraryName: String, telegramUserId: String, wordSpelling: String): Word {
+        val libraryId = libraryRepository.getLibraryIdByPairLibraryNameAndTelegramUserId(libraryName, telegramUserId)
+        val wordId = wordRepository.getWordIdBySpellingAndLibraryId(libraryName, telegramUserId)
+        wordId.takeIf {
+            wordRepository.isWordBelongsToLibraryByWordId(wordId, libraryId)
+        }?.let {
+            return wordRepository.deleteWord(wordSpelling)
+        } ?: throw WordNotFoundBySpendingException()
+    }
 
-    companion object {
-        val log = LoggerFactory.getLogger(this::class.java)
+    override fun getFullInfoAboutWord(libraryName: String, telegramUserId: String, wordSpelling: String): Word {
+        val libraryId = libraryRepository.getLibraryIdByPairLibraryNameAndTelegramUserId(libraryName, telegramUserId)
+        val wordId = wordRepository.getWordIdBySpellingAndLibraryId(wordSpelling, libraryId)
+        return wordRepository.getFullInfoAboutWord(wordId)
```

Comment:
is it possible to do this with aggregation, so there will be only one call to DB?